packages/server/graphql/mutations/helpers/generateGroupSummaries.ts

```diff
@@ -0,0 +1,33 @@
+import getRethink from '../../../database/rethinkDriver'
+import OpenAIServerManager from '../../../utils/OpenAIServerManager'
+import {DataLoaderWorker} from '../../graphql'
+
+const generateGroupSummaries = async (
+  meetingId: string,
+  dataLoader: DataLoaderWorker,
+  facilitatorUserId: string
+) => {
+  const [reflections, reflectionGroups, facilitator] = await Promise.all([
+    dataLoader.get('retroReflectionsByMeetingId').load(meetingId),
+    dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId),
+    dataLoader.get('users').loadNonNull(facilitatorUserId)
+  ])
+  if (!facilitator.featureFlags.includes('aiSummary')) return
+  const r = await getRethink()
+  const manager = new OpenAIServerManager()
+  for (const group of reflectionGroups) {
+    const reflectionsByGroupId = reflections.filter(
+      ({reflectionGroupId}) => reflectionGroupId === group.id
+    )
+    if (reflectionsByGroupId.length <= 1) continue
+    const reflectionTextByGroupId = reflectionsByGroupId.map(
+      ({plaintextContent}) => plaintextContent
+    )
+    if (reflectionTextByGroupId.length === 0) continue
+    const summary = await manager.getSummary(reflectionTextByGroupId)
+    if (!summary) continue
+    r.table('RetroReflectionGroup').get(group.id).update({summary}).run()
```

Comment:
+1 Even if we don't await from the calling side, it would be nice to have the ability to do so, i.e. we should properly await all async calls in this function. Best would be to replace the line 18 with a
```ts
return Promise.all(reflectionGroups.map(async (group) => {
```
and await or return here.